src/net.cpp

```diff
@@ -563,11 +566,17 @@ void CNode::Ban(const CSubNet& subNet, const BanReason &banReason, int64_t banti
         setBanned[subNet] = banEntry;
 
     setBannedIsDirty = true;
+
+    DumpBanlist(); //store banlist to disk
+    uiInterface.BannedListChanged();
 }
 
 bool CNode::Unban(const CNetAddr &addr) {
     CSubNet subNet(addr);
     return Unban(subNet);
+
+    DumpBanlist(); //store banlist to disk
```

Comment:
Unreachable code, called after return statement (also redundant as should be called by the subnet overload of the Unband function).